src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/auth/AuthPageContext.java

```diff
@@ -0,0 +1,56 @@
+package com.wikia.webdriver.pageobjectsfactory.pageobject.auth;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+import com.wikia.webdriver.pageobjectsfactory.componentobject.modalwindows.FacebookSignupModalComponentObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register.AttachedRegisterPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.register.RegisterPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin.AttachedSignInPage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.auth.signin.SignInPage;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class AuthPageContext extends WikiBasePageObject implements FacebookAuthContext {
+
+  @FindBy(css = ".signup-provider-facebook")
+  private WebElement facebookSignUpButton;
+  @FindBy(css = ".signup-providers li a")
+  private WebElement connectWithFacebookButton;
+  @FindBy(css = ".register-page .header-callout-link")
+  private WebElement linkToSignInForm;
+  @FindBy(css = ".signin-page .header-callout-link")
+  private WebElement linkToSignUpForm;
+  @FindBy(css = ".auth-header")
+  private WebElement authHeader;
+  @FindBy(css = ".auth-content")
+  private WebElement content;
+
+  @Override public FacebookSignupModalComponentObject clickFacebookSignUp() {
+    wait.forElementClickable(facebookSignUpButton).click();
+    PageObjectLogging.log("clickFacebookSignUp", "clicked on sign up with facebok button", true);
+    return new FacebookSignupModalComponentObject();
+  }
+
+  @Override public boolean isConnetctWithFacebookButtonVisible() {
+    return wait.forElementVisible(connectWithFacebookButton).isDisplayed();
+  }
+
+  public SignInPage navigateToSignIn() {
+    waitAndClick(linkToSignInForm);
+    return new AttachedSignInPage();
+  }
+
+  public RegisterPage navigateToSignUp() {
+    waitAndClick(linkToSignUpForm);
+    return new AttachedRegisterPage();
+  }
+
+  public boolean isHeaderDisplayed() {
+    return wait.forElementVisible(authHeader).isDisplayed();
+  }
+
+  public boolean containsText(String text) {
+    System.out.println("Text: " + wait.forElementVisible(content).getText() + " ||EOL");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS106)